### PR TITLE
Add parameter on job template level

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/46673

---

<s>TODOs:

* Use the parameters when scheduling products
* Extend regular REST-API for consistency (shouldn't be much effort)</s>

I wouldn't support a description anymore as stated in the issue (because on job template level it doesn't make too much sense).

### CODE REVIEW

File: public/schema/JobTemplate.yaml [hunk 1]
```diff
@@ -40,6 +40,13 @@ patternProperties:
                             type: string
                           priority:
                             type: number
+                          parameter:
```
Comment (coolo) [inline public/schema/JobTemplate.yaml]: I wasn't sure but the plural is parameters: https://en.wiktionary.org/wiki/parameter

Comment (Martchus) [reply public/schema/JobTemplate.yaml]: Right - I'll fix that. Or maybe just call it settings for consistency with everything else? Internally I also called the table JobTemplateSettings to be consistent with the other *Settings tables.

Comment (coolo) [reply public/schema/JobTemplate.yaml]: Go for it - I don't think it makes a difference.

**Comment by AdamWill** [comment]
Historical note: ever since this PR, the test fixtures have contained something I think is more or less impossible: a "legacy" job group ("opensuse", group id 1001) which has an associated job template containing these "job template settings".

`01-jobs.pl` has this job group:
```
    JobGroups => {
        id => 1001,
        name => 'opensuse',
        sort_order => 0,
        description => "## Test description\n\nwith bugref bsc#1234",
    },
```
note it has no `template`, so it's a legacy group by definition. `04-products.pl` defines this job template within the opensuse-13.1-DVD-i586 product:
```
            {
                machine => {name => '64bit'},
                test_suite => {name => 'advanced_kde'},
                prio => 40,
                group_id => 1001,
                settings => [{key => 'DESKTOP', value => 'advanced_kde'}, {key => 'ADVANCED', value => '1'},],
            },
```
that's a job template with a settings array, associated with group_id 1001, our legacy 'opensuse' group. This is, I think, a state it's impossible to reach with anything *other* than direct database injection. I think the only 'legitimate' way to create these job template settings is via the job group's YAML template string. They can only get in via `create_or_update_job_template` in `Schema::ResultSet::JobTemplates`, which is only called by `_update_job_templates` in `Controller::API::V1::JobTemplate`, which is only called by `_perform_update`. That in turn is only called by `update`, which is exactly the function that backs the `job_templates_scheduling` API endpoint - i.e. this whole thing is the YAML template stack, you have to pass that API function a `template`  and that `template` is going to wind up in the job group.

I'm pretty sure there's no way you can get a `settings` array into a job template via any legacy, pre-YAML way of creating job templates/job groups. The legacy web UI doesn't offer any such thing, AFAICS, and neither does the legacy(?) `job_templates` POST API endpoint, which is hooked up to the controller's `create` function, which has no ability that I can see to create these `settings`.

So...the `opensuse` group in the test fixtures really should've been turned into a modern-style group at the time this was merged.

**Pull request merged**